### PR TITLE
#655 Update to latest jmdns

### PR DESCRIPTION
**Pull request by wimjongman**
* fixed

### CODE REVIEW

**Comment by wimjongman** [comment]
Is there a way to pickup the build from Travis?

**Comment by jantje** [comment]
@wimjongman 
yes goto sloeber.io->install->latest nightly :-)
Do you want to do more testing before I merge?

**Comment by jantje** [comment]
@wimjongman 
Oeps no that is only after the merge.
I'm trying to see whether this is possible

**Comment by jantje** [comment]
@wimjongman 
I don't think these versions get uploaded anywhere.
But it compiled and tested ok
(as expected as jdmns is not involved here)
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 78.022 sec - in io.sloeber.core.RegressionTest

is_extern_C_taken_into_account(io.sloeber.core.RegressionTest)  Time elapsed: 12.18 sec

are_defines_before_includes_taken_into_account(io.sloeber.core.RegressionTest)  Time elapsed: 5.597 sec

redirectedJson(io.sloeber.core.RegressionTest)  Time elapsed: 4.359 sec

issue555(io.sloeber.core.RegressionTest)  Time elapsed: 0 sec

issue687(io.sloeber.core.RegressionTest)  Time elapsed: 4.146 sec

are_jantjes_options_taken_into_account(io.sloeber.core.RegressionTest)  Time elapsed: 3.804 sec

Results :

Tests run: 6, Failures: 0, Errors: 0, Skipped: 0

PID=3666 still running at 720 seconds

Ran for 720 seconds

Exit status of 0

File: io.sloeber.core/build.properties [hunk 1]
```diff
@@ -9,7 +9,8 @@ bin.includes = .,\
                config/,\
                OSGI-INF/,\
                src/io/sloeber/core/templates/sketch.ino,\
-               lib/
+               lib/,\
+               lib/jmdns-3.5.3.jar
```
Comment (jantje) [inline io.sloeber.core/build.properties]: As there are multiple jars there I think it is better to have them all mentioned or none. Now there is just one.
PS only saying this as you asked my review :-)

Comment (wimjongman) [reply io.sloeber.core/build.properties]: Yes thanks. That build.properties is a mess,

Comment (jantje) [reply io.sloeber.core/build.properties]: I don't feel qualified to confirm this.
I'm also thinking: The mess has probably some correlation with my sentence just before this one. ;-)

**Comment by wimjongman** [comment]
i'm running a local test and I get:

java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
	at javax.jmdns.impl.NetworkTopologyDiscoveryImpl.<clinit>(NetworkTopologyDiscoveryImpl.java:24)
	at javax.jmdns.NetworkTopologyDiscovery$Factory.newNetworkTopologyDiscovery(NetworkTopologyDiscovery.java:114)
	at javax.jmdns.NetworkTopologyDiscovery$Factory.getInstance(NetworkTopologyDiscovery.java:126)
	at cc.arduino.packages.discoverers.NetworkDiscovery.start(NetworkDiscovery.java:106)
	at io.sloeber.core.Activator$2.run(Activator.java:221)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory cannot be found by io.sloeber.core_4.1.0.201710171639
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:484)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:395)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:387)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:150)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 6 more

**Comment by wimjongman** [comment]
I fixed that error but the code in cc.arduino that we have copied in io.sloeber.core is way out of sync. I don't know what that code is supposed to do but we are using a bunch of JMDNS internal API that cc.arduino now has resolved.

theirs: https://github.com/arduino/Arduino/blob/master/arduino-core/src/cc/arduino/packages/discoverers/NetworkDiscovery.java

ours: https://github.com/Sloeber/arduino-eclipse-plugin/blob/master/io.sloeber.core/src/cc/arduino/packages/discoverers/NetworkDiscovery.java

**Comment by jantje** [comment]
I took a look. We should be able to copy the file without issues.
I guess you prefer me doing so as I see some files are no longer part of arduino.cc

**Comment by jantje** [comment]
@wimjongman I propose to accept this change and then I'll upgrade the arduino.cc package

**Pull request merged**